src/everest/everest_storage.py

```diff
@@ -0,0 +1,893 @@
+                        1.0 / s for s in config.nonlinear_constraints.scales
+                    ],  # Q: Is this correct?
+                    "constraint_rhs_value": config.nonlinear_constraints.rhs_values,
+                    "constraint_type": config.nonlinear_constraints.types,
+                }
+            )
+
+        self.data.realization_weights = polars.DataFrame(
+            {
+                "realization": polars.Series(
+                    config.realizations.names, dtype=polars.UInt32
+                ),
+                "weight": polars.Series(
+                    config.realizations.weights, dtype=polars.Float64
+                ),
+            }
+        )
+
+    def _store_function_results(self, results: FunctionResults) -> _EvaluationResults:
+        # We could select only objective values,
+        # but we select all to also get the constraint values (if they exist)
+        realization_objectives = polars.from_pandas(
+            results.to_dataframe(
+                "evaluations",
+                select=["objectives", "evaluation_ids"],
+            ).reset_index(),
+        ).select(
+            "batch_id",
+            "realization",
+            "objective",
+            "objectives",
+            "evaluation_ids",
+        )
+
+        if results.nonlinear_constraints is not None:
+            realization_constraints = polars.from_pandas(
+                results.to_dataframe(
+                    "evaluations",
+                    select=["constraints", "evaluation_ids"],
+                ).reset_index(),
+            ).select(
+                "batch_id",
+                "realization",
+                "evaluation_ids",
+                "nonlinear_constraint",
+                "constraints",
+            )
+
+            realization_constraints = self._rename_ropt_df_columns(
+                realization_constraints
+            )
+
+            batch_constraints = polars.from_pandas(
+                results.to_dataframe("nonlinear_constraints").reset_index()
+            ).select("batch_id", "nonlinear_constraint", "values", "violations")
+
+            batch_constraints = batch_constraints.rename(
+                {
+                    "nonlinear_constraint": "constraint_name",
+                    "values": "constraint_value",
+                    "violations": "constraint_violation",
+                }
+            )
+
+            constraint_names = batch_constraints["constraint_name"].unique().to_list()
+
+            batch_constraints = batch_constraints.pivot(
+                on="constraint_name",
+                values=[
+                    "constraint_value",
+                    "constraint_violation",
+                ],
+                separator=";",
+            ).rename(
+                {
+                    **{f"constraint_value;{name}": name for name in constraint_names},
+                    **{
+                        f"constraint_violation;{name}": f"{name}.violation"
```

Comment:
I think it would be better not to access the `violation` fields at the moment. They are used by `ropt` to report constraint violations, but that is not used by Everest. I think I will make some changes to `ropt` in the way these are reported, so lets not use this for now.